integration_tests/src/main/python/hybrid_parquet_test.py

```diff
@@ -290,6 +290,413 @@ def test_hybrid_parquet_filter_pushdown_timestamp(spark_tmp_path):
+    "(get_json_object(json_str, '$.a') == 'a')",
+    "(map_values(map(int_not_null, str1)) == array('a'))",
+    "(if(bool1, 1, 2) == 1)",
+    pytest.param("(1 in(int1, int2, int3))", marks=pytest.mark.xfail(reason='not supported by gluten')),
+    pytest.param("(json_array_length(json_str) == 1)", marks=pytest.mark.xfail(reason='not supported by gluten')),
+    "(like(str1, '_e_'))",
+    "(isnotnull(map_from_arrays(array(int1, int2), array(str1, str2))))",
+    "(map_keys(map(int_not_null, str1)) == array(1))",
+    "(map_values(map(int_not_null, str1)) == array('b'))",
+    "(isnotnull(map_zip_with(map(int1, str1, int2, str2), map(int3, str3), (k, v1, v2) -> if(k == int3, v2, v1))))",
+    "(next_day(date1, 'monday') == date1)",
+    "(overlay(str1, 'a', 1) == 'hallo')",
+    "(isnotnull(str_to_map(str1)))",
+    "(substring_index(str1, 'e', 2) == 'he')",
+    pytest.param("(to_unix_timestamp(timestamp) == 1717171717)", marks=pytest.mark.xfail(reason='timestamp filter not supported by gluten')),
+    "(unhex('68656c6c6f') == 'hello')",
+    # Ansi false only:
+    # Multiply, Add, Subtract, Divide, Abs, Pmod
+    "(int1 * int2 == 24)",
+    "(int1 + int2 == 5)",
+    "(int1 - int2 == 1)",
+    "(int1 / int2 == 2)",
+    "(abs(int1) == 1)",
+    "(pmod(double1, double2) == 1)",
+]
+
+@pytest.mark.skipif(is_databricks_runtime(), reason="Hybrid feature does not support Databricks currently")
+@pytest.mark.skipif(not is_hybrid_backend_loaded(), reason="HybridScan specialized tests")
+@hybrid_test
+@pytest.mark.parametrize('condition', condition_list, ids=idfn)
+def test_hybrid_parquet_filter_pushdown_more_exprs(spark_tmp_path, condition):
+    data_path = spark_tmp_path + '/PARQUET_DATA'
+    genlist = [('bool1', BooleanGen()),
+               ('bool2', BooleanGen()),
+               ('double1', DoubleGen()),
+               ('double2', DoubleGen()),
+               ('double3', DoubleGen()),
+               ('int1', IntegerGen()),
+               ('int2', IntegerGen()),
+               ('int3', IntegerGen()),
+               ('int4', IntegerGen()),
+               ('int_not_null', IntegerGen(nullable=False)),
+               ('date1', DateGen()),
+               ('date2', DateGen()),
+               ('timestamp', TimestampGen(start=datetime(1900, 1, 1, tzinfo=timezone.utc))),
+               ('array_int_1', ArrayGen(IntegerGen())),
+               ('array_int_2', ArrayGen(IntegerGen())),
+               ('array_array_int_1', ArrayGen(ArrayGen(IntegerGen(),max_length=3))),
+               ('array_str_1', ArrayGen(StringGen())),
+               ('str1', StringGen()),
+               ('str2', StringGen()),
+               ('str3', StringGen()),
+               ('digitstr', StringGen(pattern='[0-9]{1,5}')),
+               ('long', LongGen()),
+               ('comma_separated_str', StringGen(pattern='([0-9],){1,5}')),
+               ('json_str', StringGen(pattern=r'\{"a": "[a-z]{1,3}"\}')),
+    ]
+    with_cpu_session(
```

Comment:
Just NIT: I think the tmp test file can be somehow reused since the data schema is unchanged among all conditions. Perhaps the reuse can save a lot of time?

Reply:
Good point!

Now I played a trick to filter out unneeded data gens to make it run faster.

Alternatively we can:

1. combine conditions with and (or xor?) to reduce the number of test cases. But I'd like to see each expression tested separately so we can quickly locate the problem if something fails. Especially if people are using a custom gluten jar, they may want to run these tests to find out a supported list for themselves.
2. Introduce some caching for files on spark_temp_path, like we did for data frames in integration tests. So we can speed up all cases with `spark_tmp_path`, there are many of them. I need to think more about this approach...

Reply:
after the datagen filetering, time of `test_hybrid_parquet_filter_pushdown` reduced from 2:15 to 1:50

Reply:
Perhaps we can just leverage `pytest.fixture`? Although it can be shared across parallel workers launched by pytest.

Reply:
Thanks! `pytest.fixture` looks good and I tried but got slower `158.41s (0:02:38)`, not sure why.
```
@pytest.fixture(scope="function")
def prepared_parquet_data(spark_tmp_path):
    """Create parquet test data for filter pushdown tests."""
    data_path = spark_tmp_path + '/PARQUET_DATA'
    
    gen_list = filter_pushdown_gen_list

    with_cpu_session(
        lambda spark: gen_df(spark, gen_list).write.parquet(data_path),
        conf=rebase_write_corrected_conf)
    
    return data_path
```